nautobot/dcim/forms.py

```diff
@@ -886,48 +898,158 @@ class DeviceTypeFilterForm(NautobotFilterForm):
+    )
+    has_console_ports = forms.NullBooleanField(
+        required=False,
+        label="Has console ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_console_server_ports = forms.NullBooleanField(
+        required=False,
+        label="Has console server ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_power_ports = forms.NullBooleanField(
+        required=False,
+        label="Has power ports",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_power_outlets = forms.NullBooleanField(
+        required=False,
+        label="Has power outlets",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    has_interfaces = forms.NullBooleanField(
+        required=False,
+        label="Has interfaces",
+        widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES),
+    )
+    tags = TagFilterField(model)
+
+
 #
 # Device component templates
 #
 
 
+class ComponentTemplateForm(NautobotModelForm):
+    # TODO: placeholder values shouldn't be form controls, instead use:
+    #       <p class="form-control-static">{{ obj|hyperlinked_object_target_new_tab }}</p>
+    device_type = DynamicModelChoiceField(
+        queryset=DeviceType.objects.all(),
+    )
+
+
+class ModularComponentTemplateForm(ComponentTemplateForm):
+    device_type = DynamicModelChoiceField(
+        queryset=DeviceType.objects.all(),
+        required=False,
+    )
+    module_type = DynamicModelChoiceField(
+        queryset=ModuleType.objects.all(),
+        required=False,
+    )
+
+
 class ComponentTemplateCreateForm(ComponentForm):
     """
     Base form for the creation of device component templates (subclassed from ComponentTemplateModel).
     """
 
-    manufacturer = DynamicModelChoiceField(
-        queryset=Manufacturer.objects.all(),
-        required=False,
-        initial_params={"device_types": "device_type"},
-    )
     device_type = DynamicModelChoiceField(
         queryset=DeviceType.objects.all(),
-        query_params={"manufacturer": "$manufacturer"},
     )
     description = forms.CharField(required=False)
 
 
-class ConsolePortTemplateForm(NautobotModelForm):
+class ModularComponentTemplateCreateForm(ComponentTemplateCreateForm):
+    """
+    Base form for the creation of modular device component templates (subclassed from ModularComponentTemplateModel).
+    """
+
+    device_type = DynamicModelChoiceField(
+        queryset=DeviceType.objects.all(),
+        required=False,
+    )
+    module_type = DynamicModelChoiceField(
+        queryset=ModuleType.objects.all(),
+        required=False,
+    )
+
+
+class ConsolePortTemplateForm(ModularComponentTemplateForm):
     class Meta:
         model = ConsolePortTemplate
         fields = [
             "device_type",
+            "module_type",
             "name",
             "label",
             "type",
             "description",
         ]
```

Comment:
Not blocking, but feels like we could cut down on boilerplate a fair bit by declaring `fields` on `ModularComponentTemplateForm.Meta` and `ModularComponentTemplateCreateForm.Meta` and having the concrete subclasses `Meta` inherit from those, or at least do something like `fields = [*ModularComponentTemplateForm.Meta.fields, ...]`?